src/sentry/api/endpoints/event_file_committers.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+import six
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import (
+    Release, ReleaseCommit, Commit, CommitFileChange, Event
+)
+from sentry.api.serializers.models.release import get_users_for_commits
+
+from itertools import izip
+from collections import defaultdict
+
+
+def tokenize_path(path):
+    # TODO(maxbittker) tokenize in a smarter crossplatform way.
+    return reversed(path.split('/'))
+
+
+class EventFileCommittersEndpoint(ProjectEndpoint):
+
+    def _get_frame_paths(self, event):
+        data = event.data
+        try:
+            frames = data['sentry.interfaces.Stacktrace']['frames']
+        except KeyError:
+            try:
+                frames = data['sentry.interfaces.Exception']['values'][0]['stacktrace']['frames']
+            except KeyError:
+                return []  # can't find stacktrace information
+
+        return frames
+
+    def _score_path_match_length(self, path_a, path_b):
+        score = 0
+        for a, b in izip(tokenize_path(path_a), tokenize_path(path_b)):
+            if a != b:
+                break
+            score += 1
+        return score
+
+    def _get_commits(self, project_id, version):
+        try:
+            commits = Commit.objects.filter(
+                releasecommit=ReleaseCommit.objects.filter(
+                    release=Release.objects.get(
+                        projects=project_id,
+                        version=version,
+                    ),
+                )
+            )
+        except Release.DoesNotExist:
```

Comment:
https://travis-ci.org/getsentry/sentry/jobs/215716211#L1223 this actually does happen in the tests, because of the get, not the filter. can I keep this in or would you rather I do a filter and first so that it returns an empty list?